### PR TITLE
dirent.h dependency for windows removed

### PR DESCRIPTION
**Pull request by m4b4**
dirent.h is removed now
apprentice_load in apprentice.c is and fs_posix.c are not tested yet
I am also thinking about a function that returns a HANDLE and a struct that contains the variables (dir, wcpath, cfname, buffer)needed for FindFirstFileW to make the code more readable

### CODE REVIEW

File: libr/magic/apprentice.c [hunk 1]
```diff
@@ -36,7 +36,11 @@
 #include <r_util.h>
 #include <sys/param.h>
 #include <ctype.h>
+#if (__WINDOWS__ && !defined(__CYGWIN__))
```
Comment (radare) [inline libr/magic/apprentice.c]: remove this block, dirent its already in r_util

File: libr/util/sandbox.c [hunk 2]
```diff
@@ -277,7 +277,23 @@ R_API int r_sandbox_kill(int pid, int sig) {
 #endif
 	return -1;
 }
+#if __WINDOWS__ && !defined(__CYGWIN__)
+R_API HANDLE r_sandbox_opendir (const char *path, WIN32_FIND_DATAW *entry, wchar_t *dir, wchar_t *wcpath) {
```
Comment (radare) [inline libr/util/sandbox.c]: i think function signature should not change depending on platform.

File: libr/util/sandbox.c [hunk 3]
```diff
@@ -277,7 +277,23 @@ R_API int r_sandbox_kill(int pid, int sig) {
...
+	if (!path)
```
Comment (radare) [inline libr/util/sandbox.c]: missing braces

File: libr/util/sys.c [hunk 4]
```diff
@@ -122,6 +121,34 @@ R_API int r_sys_truncate(const char *file, int sz) {
...
 R_API RList *r_sys_dir(const char *path) {
 	RList *list = NULL;
+#if __WINDOWS__ && !defined(__CYGWIN__)
+	HANDLE fh;
+	WIN32_FIND_DATAW entry;
+	wchar_t dir[MAX_PATH];
+	wchar_t *wcpath;
+	char *cfname;
+
+	fh = r_sandbox_opendir (path, &entry, &dir, wcpath);
```
Comment (radare) [inline libr/util/sys.c]: oh wait... handle here.. this can make the whole thing a bit tricky if we want to keep function signatures across platforms

Comment (m4b4) [inline libr/util/sys.c]: yep.. idk how to handle that, do you have any idea?

File: libr/util/sys.c [hunk 5]
```diff
@@ -122,6 +121,34 @@ R_API int r_sys_truncate(const char *file, int sz) {
...
+
+	fh = r_sandbox_opendir (path, &entry, dir, wcpath);
+	if (fh == INVALID_HANDLE_VALUE) {
+		//IFDGB eprintf ("Cannot open directory %ls\n", wcpath);
+		if (wcpath) {
+			free (wcpath);
+		}
+		return list;
+	}
+	list = r_list_new ();
```
Comment (radare) [inline libr/util/sys.c]: use r_list_newf instead of later assigning the ->free pointer on every iteration

File: libr/util/sys.c [hunk 6]
```diff
@@ -122,6 +121,34 @@ R_API int r_sys_truncate(const char *file, int sz) {
...
+	HANDLE fh;
+	WIN32_FIND_DATAW entry;
+	wchar_t dir[MAX_PATH];
+	wchar_t *wcpath;
+	char *cfname;
+
+	fh = r_sandbox_opendir (path, &entry, dir, wcpath);
+	if (fh == INVALID_HANDLE_VALUE) {
+		//IFDGB eprintf ("Cannot open directory %ls\n", wcpath);
+		if (wcpath) {
```
Comment (radare) [inline libr/util/sys.c]: this is unnecessary, free already checks for null

**Comment by radare** [comment]
Thanks!

**Pull request merged**